### PR TITLE
CNI: Update flannel from v0.20.2 to v0.21.4 and change repository

### PR DESCRIPTION
**Pull request by zhengtianbao**
flannel has moved its images to: docker.io we should switch to docker.io/flannel and update to the latest version.
stay up to date with upstream:  https://github.com/flannel-io/flannel/blob/master/Documentation/kube-flannel.yml

### CODE REVIEW

File: pkg/minikube/cni/flannel.go [hunk 1]
```diff
@@ -51,12 +50,6 @@ func (c Flannel) String() string {
 
 // Apply enables the CNI
 func (c Flannel) Apply(r Runner) error {
-	// Mostly applicable to the 'none' driver
-	_, err := r.RunCmd(exec.Command("stat", "/opt/cni/bin/portmap"))
-	if err != nil {
-		return errors.Wrap(err, "required 'portmap' CNI plug-in not found")
```
Comment (spowelljr) [inline pkg/minikube/cni/flannel.go:57]: Is `portmap` no longer required for flannel?

Comment (afbjorklund) [reply pkg/minikube/cni/flannel.go:57]: It is still required, as part of the CNI config

File: pkg/minikube/cni/flannel.yaml [hunk 2]
```diff
@@ -168,9 +168,6 @@ spec:
           requests:
             cpu: "100m"
             memory: "50Mi"
-          limits:
```
Comment (afbjorklund) [inline pkg/minikube/cni/flannel.yaml:171]: This change also seems unrelated.

**Review by afbjorklund — CHANGES_REQUESTED** [review]

File: pkg/minikube/cni/flannel.yaml (hunk 2 above)
Comment (afbjorklund) [reply pkg/minikube/cni/flannel.yaml:171]: It is from upstream ([30b56a37](https://github.com/flannel-io/flannel/commit/30b56a3794737135bf69d983417dcaaf3f1798b8)), but should probably be mentioned separately.

File: pkg/minikube/cni/flannel.go (hunk 1 above)
Comment (zhengtianbao) [reply pkg/minikube/cni/flannel.go]: ok, the error may occur, keep these lines.

File: pkg/minikube/cni/flannel.yaml (hunk 2 above)
Comment (zhengtianbao) [reply pkg/minikube/cni/flannel.yaml]: reverted.

File: pkg/minikube/cni/flannel.yaml [hunk 3]
```diff
@@ -130,8 +130,8 @@ spec:
       serviceAccountName: flannel
       initContainers:
       - name: install-cni-plugin
-        #image: flannelcni/flannel-cni-plugin:v1.1.2 #for ppc64le and mips64le (dockerhub limitations may apply)
-        image: docker.io/rancher/mirrored-flannelcni-flannel-cni-plugin:v1.1.2
+        image: docker.io/flannel/flannel-cni-plugin:v1.1.2
+        #image: docker.io/rancher/mirrored-flannelcni-flannel-cni-plugin:v1.1.2
```
Comment (medyagh) [inline pkg/minikube/cni/flannel.yaml:134]: why keeping the old one in comment ? we could remove the un-used deprecated image

**Review by medyagh — CHANGES_REQUESTED** [review]

File: pkg/minikube/cni/flannel.yaml (hunk 3 above)
Comment (spowelljr) [reply pkg/minikube/cni/flannel.yaml:134]: I believe it's commented in the upstream file, seems like a backup if the main one is down

**Review by spowelljr — APPROVED** [review]
